lib/opal/nodes/args/initialize_kwargs.rb

```diff
@@ -0,0 +1,32 @@
+require 'opal/nodes/base'
+
+module Opal
+  module Nodes
+    # An abstract node responsible for validating
+    # keyword arguments in the post-splat list of arguments.
+    #
+    # @see PostArgsNode
+    #
+    class InitializeKwargsNode < Base
+      def compile
+        unless scope.kwargs_initialized
+          initialize_kw_args
+        end
+      end
+
+      def initialize_kw_args
```

Comment:
or maybe removing `#compile` altogether and call `#initialize_kw_args` explicitly in subclasses instead of `super`, this I think would make more clear what's going on.